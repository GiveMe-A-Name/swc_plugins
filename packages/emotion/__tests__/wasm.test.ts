import { describe, expect, test } from "vitest";
import path from "node:path";
import fs from "node:fs/promises";
import url from "node:url";
import { type Options, transform } from "@swc/core";

const pluginName = "swc_plugin_emotion.wasm";

const options: Options = {
  jsc: {
    parser: {
      syntax: "ecmascript",
    },
    experimental: {
      plugins: [
        [
          path.join(
            path.dirname(url.fileURLToPath(import.meta.url)),
            "..",
            pluginName,
          ),
          {},
        ],
      ],
    },
  },
};

test("Should transform emotion css correctly", async () => {
  const code = await fs.readFile(
    path.resolve(
      url.fileURLToPath(import.meta.url),
      "..",
      "fixtures",
      "input.js",
    ),
    "utf-8",
  );
  const output = await transform(code, options);
  expect(output.code).toMatchSnapshot();
});
