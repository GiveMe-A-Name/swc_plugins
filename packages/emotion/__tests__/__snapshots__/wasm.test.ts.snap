// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should transform emotion css correctly 1`] = `
"function _tagged_template_literal(strings, raw) {
    if (!raw) {
        raw = strings.slice(0);
    }
    return Object.freeze(Object.defineProperties(strings, {
        raw: {
            value: Object.freeze(raw)
        }
    }));
}
function _templateObject() {
    var data = _tagged_template_literal([
        "\\n      width: 96px;\\n      height: 96px;\\n    "
    ]);
    _templateObject = function _templateObject() {
        return data;
    };
    return data;
}
function _templateObject1() {
    var data = _tagged_template_literal([
        "\\n  font-size: 20px;\\n  @media (min-width: 420px) {\\n    color: blue;\\n    ",
        ";\\n    line-height: 26px;\\n  }\\n  background: green;\\n  ",
        ";\\n"
    ]);
    _templateObject1 = function _templateObject() {
        return data;
    };
    return data;
}
var myStyles = css(_templateObject1(), css(_templateObject()), {
    backgroundColor: "hotpink"
});
"
`;
